
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>basenji_data &#8212; Init_website 1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css" />
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="module-basenji_data">
<span id="basenji-data"></span><h1><a class="reference internal" href="#module-basenji_data" title="basenji_data"><code class="xref py py-mod docutils literal notranslate"><span class="pre">basenji_data</span></code></a><a class="headerlink" href="#module-basenji_data" title="Permalink to this heading">¶</a></h1>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this heading">¶</a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this heading">¶</a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.main" title="basenji_data.main"><code class="xref py py-obj docutils literal notranslate"><span class="pre">main</span></code></a>()</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.annotate_unmap" title="basenji_data.annotate_unmap"><code class="xref py py-obj docutils literal notranslate"><span class="pre">annotate_unmap</span></code></a>(mseqs, unmap_bed, seq_length, pool_width)</p></td>
<td><p>Intersect the sequence segments with unmappable regions</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.break_large_contigs" title="basenji_data.break_large_contigs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">break_large_contigs</span></code></a>(contigs, break_t[, verbose])</p></td>
<td><p>Break large contigs in half until all contigs are under</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.contig_sequences" title="basenji_data.contig_sequences"><code class="xref py py-obj docutils literal notranslate"><span class="pre">contig_sequences</span></code></a>(contigs, seq_length, stride[, snap, ...])</p></td>
<td><p>Break up a list of Contig's into a list of ModelSeq's.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.curate_peaks" title="basenji_data.curate_peaks"><code class="xref py py-obj docutils literal notranslate"><span class="pre">curate_peaks</span></code></a>(targets_df, out_dir, pool_width, crop_bp)</p></td>
<td><p>Merge all peaks, round to nearest pool_width, and add cropped bp.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.divide_contigs_chr" title="basenji_data.divide_contigs_chr"><code class="xref py py-obj docutils literal notranslate"><span class="pre">divide_contigs_chr</span></code></a>(contigs, test_chrs, valid_chrs)</p></td>
<td><p>Divide list of contigs into train/valid/test lists</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.divide_contigs_folds" title="basenji_data.divide_contigs_folds"><code class="xref py py-obj docutils literal notranslate"><span class="pre">divide_contigs_folds</span></code></a>(contigs, folds)</p></td>
<td><p>Divide list of contigs into cross fold lists.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.divide_contigs_pct" title="basenji_data.divide_contigs_pct"><code class="xref py py-obj docutils literal notranslate"><span class="pre">divide_contigs_pct</span></code></a>(contigs, test_pct, valid_pct[, ...])</p></td>
<td><p>Divide list of contigs into train/valid/test lists,</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.limit_contigs" title="basenji_data.limit_contigs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">limit_contigs</span></code></a>(contigs, filter_bed)</p></td>
<td><p>Limit to contigs overlapping the given BED.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.rejoin_large_contigs" title="basenji_data.rejoin_large_contigs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">rejoin_large_contigs</span></code></a>(contigs)</p></td>
<td><p>Rejoin large contigs that were broken up before alignment comparison.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.write_seqs_bed" title="basenji_data.write_seqs_bed"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_seqs_bed</span></code></a>(bed_file, seqs[, labels])</p></td>
<td><p>Write sequences to BED file.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="attributes">
<h3>Attributes<a class="headerlink" href="#attributes" title="Permalink to this heading">¶</a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_data.Contig" title="basenji_data.Contig"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Contig</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_data.ModelSeq" title="basenji_data.ModelSeq"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ModelSeq</span></code></a></p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.main">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">main</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.main" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.annotate_unmap">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">annotate_unmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mseqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unmap_bed</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_length</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pool_width</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.annotate_unmap" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Intersect the sequence segments with unmappable regions</dt><dd><p>and annoate the segments as NaN to possible be ignored.</p>
</dd>
<dt>Args:</dt><dd><p>mseqs: list of ModelSeq’s
unmap_bed: unmappable regions BED file
seq_length: sequence length (after cropping)
pool_width: pooled bin width</p>
</dd>
<dt>Returns:</dt><dd><p>seqs_unmap: NxL binary NA indicators</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.break_large_contigs">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">break_large_contigs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">break_t</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.break_large_contigs" title="Permalink to this definition">¶</a></dt>
<dd><p>Break large contigs in half until all contigs are under
the size threshold.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.contig_sequences">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">contig_sequences</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_length</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stride</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.contig_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Break up a list of Contig’s into a list of ModelSeq’s.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.curate_peaks">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">curate_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">targets_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pool_width</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">crop_bp</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.curate_peaks" title="Permalink to this definition">¶</a></dt>
<dd><p>Merge all peaks, round to nearest pool_width, and add cropped bp.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.divide_contigs_chr">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">divide_contigs_chr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">test_chrs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">valid_chrs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.divide_contigs_chr" title="Permalink to this definition">¶</a></dt>
<dd><p>Divide list of contigs into train/valid/test lists
by chromosome.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.divide_contigs_folds">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">divide_contigs_folds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">folds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.divide_contigs_folds" title="Permalink to this definition">¶</a></dt>
<dd><p>Divide list of contigs into cross fold lists.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.divide_contigs_pct">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">divide_contigs_pct</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">test_pct</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">valid_pct</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pct_abstain</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.divide_contigs_pct" title="Permalink to this definition">¶</a></dt>
<dd><p>Divide list of contigs into train/valid/test lists,
aiming for the specified nucleotide percentages.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.limit_contigs">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">limit_contigs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_bed</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.limit_contigs" title="Permalink to this definition">¶</a></dt>
<dd><p>Limit to contigs overlapping the given BED.</p>
<dl class="simple">
<dt>Args</dt><dd><p>contigs: list of Contigs
filter_bed: BED file to filter by</p>
</dd>
<dt>Returns:</dt><dd><p>fcontigs: list of Contigs</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.rejoin_large_contigs">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">rejoin_large_contigs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.rejoin_large_contigs" title="Permalink to this definition">¶</a></dt>
<dd><p>Rejoin large contigs that were broken up before alignment comparison.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_data.write_seqs_bed">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">write_seqs_bed</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bed_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">labels</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_data.write_seqs_bed" title="Permalink to this definition">¶</a></dt>
<dd><p>Write sequences to BED file.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="basenji_data.Contig">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">Contig</span></span><a class="headerlink" href="#basenji_data.Contig" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="basenji_data.ModelSeq">
<span class="sig-prename descclassname"><span class="pre">basenji_data.</span></span><span class="sig-name descname"><span class="pre">ModelSeq</span></span><a class="headerlink" href="#basenji_data.ModelSeq" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../index.html">Init_website</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../index.html">API Reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, David Kelley, Gunalan Natesan.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 6.1.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../_sources/autoapi/basenji_data/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>