
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>bam_cov &#8212; Init_website 1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css" />
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="module-bam_cov">
<span id="bam-cov"></span><h1><a class="reference internal" href="#module-bam_cov" title="bam_cov"><code class="xref py py-mod docutils literal notranslate"><span class="pre">bam_cov</span></code></a><a class="headerlink" href="#module-bam_cov" title="Permalink to this heading">¶</a></h1>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this heading">¶</a></h2>
<section id="classes">
<h3>Classes<a class="headerlink" href="#classes" title="Permalink to this heading">¶</a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.GenomeCoverage" title="bam_cov.GenomeCoverage"><code class="xref py py-obj docutils literal notranslate"><span class="pre">GenomeCoverage</span></code></a></p></td>
<td><p>chrom_lengths (OrderedDict): Mapping chromosome names to lengths.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this heading">¶</a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.main" title="bam_cov.main"><code class="xref py py-obj docutils literal notranslate"><span class="pre">main</span></code></a>()</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.compute_cut_norms" title="bam_cov.compute_cut_norms"><code class="xref py py-obj docutils literal notranslate"><span class="pre">compute_cut_norms</span></code></a>(cut_bias_kmer, read_weights, ...)</p></td>
<td><p>Compute cut bias normalizations.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.row_nzcols_geti" title="bam_cov.row_nzcols_geti"><code class="xref py py-obj docutils literal notranslate"><span class="pre">row_nzcols_geti</span></code></a>(m, ri)</p></td>
<td><p>Return row ri's nonzero columns.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.row_nzcols_get" title="bam_cov.row_nzcols_get"><code class="xref py py-obj docutils literal notranslate"><span class="pre">row_nzcols_get</span></code></a>(m, ri)</p></td>
<td><p>Return row ri's nonzero columns.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.row_nzcols_set" title="bam_cov.row_nzcols_set"><code class="xref py py-obj docutils literal notranslate"><span class="pre">row_nzcols_set</span></code></a>(m, ri, v)</p></td>
<td><p>Set row ri's nonzero columsn to v.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.single_or_pair" title="bam_cov.single_or_pair"><code class="xref py py-obj docutils literal notranslate"><span class="pre">single_or_pair</span></code></a>(bam_file)</p></td>
<td><p>Check the first read to guess if the BAM has single or paired end reads.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.distribute_multi_succint" title="bam_cov.distribute_multi_succint"><code class="xref py py-obj docutils literal notranslate"><span class="pre">distribute_multi_succint</span></code></a>(multi_weight_matrix, ...[, ...])</p></td>
<td><p>Wondering if I can speed things up by vectorizing these operations, but still much to test.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.initialize_kmers" title="bam_cov.initialize_kmers"><code class="xref py py-obj docutils literal notranslate"><span class="pre">initialize_kmers</span></code></a>(k[, pseudocount])</p></td>
<td><p>Initialize a dict mapping kmers to cut counts.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.regplot_gc" title="bam_cov.regplot_gc"><code class="xref py py-obj docutils literal notranslate"><span class="pre">regplot_gc</span></code></a>(vals1, vals2, model, out_pdf)</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.regplot_shift" title="bam_cov.regplot_shift"><code class="xref py py-obj docutils literal notranslate"><span class="pre">regplot_shift</span></code></a>(vals1, vals2, preds2, out_pdf)</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#bam_cov.cigar_len" title="bam_cov.cigar_len"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cigar_len</span></code></a>(cigar_str)</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#bam_cov.scatter_lims" title="bam_cov.scatter_lims"><code class="xref py py-obj docutils literal notranslate"><span class="pre">scatter_lims</span></code></a>(vals1[, vals2, buffer])</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.main">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">main</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.main" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.compute_cut_norms">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">compute_cut_norms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cut_bias_kmer</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_weights</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chromosomes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_lengths</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fasta_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.compute_cut_norms" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute cut bias normalizations.</p>
<blockquote>
<div><p>UNFINISHED!</p>
</div></blockquote>
<dl>
<dt>Args:</dt><dd><p>cut_bias_kmer
read_weights
chromosomes
chrom_lengths
fasta_file</p>
<dl class="simple">
<dt>Returns:</dt><dd><p>kmer_norms</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.row_nzcols_geti">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">row_nzcols_geti</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ri</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.row_nzcols_geti" title="Permalink to this definition">¶</a></dt>
<dd><p>Return row ri’s nonzero columns.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.row_nzcols_get">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">row_nzcols_get</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ri</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.row_nzcols_get" title="Permalink to this definition">¶</a></dt>
<dd><p>Return row ri’s nonzero columns.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.row_nzcols_set">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">row_nzcols_set</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ri</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">v</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.row_nzcols_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Set row ri’s nonzero columsn to v.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.single_or_pair">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">single_or_pair</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.single_or_pair" title="Permalink to this definition">¶</a></dt>
<dd><p>Check the first read to guess if the BAM has single or paired end reads.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.distribute_multi_succint">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">distribute_multi_succint</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">multi_weight_matrix</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_unique_coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_window</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_lengths</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_iterations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">converge_t</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.distribute_multi_succint" title="Permalink to this definition">¶</a></dt>
<dd><p>Wondering if I can speed things up by vectorizing these operations, but still much to test.</p>
<ol class="arabic simple">
<li><p>start by comparing the times to make my genome_coverage estimate here with multi_weight_matrix.sum(axis=0) versus looping through the arrays myself.</p></li>
<li><p>then benchmark that dot product.</p></li>
<li><p>then benchmark the column normalization.</p></li>
<li><p>then determine a way to assess convergence. maybe sample 1k reads to act as my proxy?</p></li>
</ol>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.initialize_kmers">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">initialize_kmers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">k</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudocount</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.initialize_kmers" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialize a dict mapping kmers to cut counts.</p>
<dl class="simple">
<dt>Args</dt><dd><p>k (int): k-mer size
pseudocount (float): Initial pseudocount</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.regplot_gc">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">regplot_gc</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vals1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vals2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_pdf</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.regplot_gc" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.regplot_shift">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">regplot_shift</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vals1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vals2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">preds2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_pdf</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.regplot_shift" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">GenomeCoverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_lengths</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stranded</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smooth_multi_sd</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clip_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clip_max_multi</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">shift_center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">shift_forward</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">shift_reverse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_overlap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_unique</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maps_t</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fasta_file</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage" title="Permalink to this definition">¶</a></dt>
<dd><p>chrom_lengths (OrderedDict): Mapping chromosome names to lengths.
fasta (pysam Fastafile):</p>
<p>unique_counts: G (genomic position) array of unique coverage counts.
multi_weight_matrix: R (reads) x G (genomic position) array of
multi-mapping read weight.
smooth_multi_sd (int): Multi-map distribution gaussian filter standard deviation.
multi_max (int): Maximum coverage per position from multi-mappers due to
mis-mapping fear.
shift (int): Alignment shift to maximize cross-strand coverage correlation</p>
<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.align_shifts">
<span class="sig-name descname"><span class="pre">align_shifts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">align</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.align_shifts" title="Permalink to this definition">¶</a></dt>
<dd><p>Helper function to determine alignment event position shifts.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.distribute_multi">
<span class="sig-name descname"><span class="pre">distribute_multi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">max_iterations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">converge_t</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.distribute_multi" title="Permalink to this definition">¶</a></dt>
<dd><p>Distribute multi-mapping read weight proportional to coverage in a local window.</p>
<dl class="simple">
<dt>In</dt><dd><p>max_iterations: Maximum iterations through the reads.
converge_t: Per read weight difference below which we consider
convergence.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.estimate_coverage">
<span class="sig-name descname"><span class="pre">estimate_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genome_coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudocount</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.estimate_coverage" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate smoothed genomic coverage.</p>
<dl class="simple">
<dt>In</dt><dd><p>genome_coverage: G (genomic position) array of estimated coverage
counts.
pseudocount (int): Coverage pseudocount.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.gc_normalize">
<span class="sig-name descname"><span class="pre">gc_normalize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">coverage</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.gc_normalize" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a model to normalize for GC content.</p>
<dl class="simple">
<dt>In</dt><dd><p>chrom (str): Chromosome
coverage ([float]): Coverage array
pseudocount (int): Coverage pseudocount.</p>
</dd>
<dt>Out</dt><dd><p>model (sklearn object): To control for GC%.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.learn_gc">
<span class="sig-name descname"><span class="pre">learn_gc</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fragment_sd</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">64</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudocount</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.learn_gc" title="Permalink to this definition">¶</a></dt>
<dd><p>Learn a model to normalize for GC content.</p>
<dl class="simple">
<dt>In</dt><dd><p>fragment_sd (int): Gaussian filter standard deviation.
pseudocount (int): Coverage pseudocount.</p>
</dd>
<dt>Out</dt><dd><p>model (sklearn object): To control for GC%.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.learn_gc_base">
<span class="sig-name descname"><span class="pre">learn_gc_base</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.learn_gc_base" title="Permalink to this definition">¶</a></dt>
<dd><p>Determine the genome-wide GC model baseline</p>
<dl class="simple">
<dt>In</dt><dd><p>self.gc_model</p>
</dd>
<dt>Out</dt><dd><p>self.gc_base</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.index_genome">
<span class="sig-name descname"><span class="pre">index_genome</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gi</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.index_genome" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute chromosome and position for a genome index.</p>
<dl class="simple">
<dt>Args</dt><dd><p>gi (int): Genomic index</p>
</dd>
<dt>Returns:</dt><dd><p>ci (int): Chromosome index.
pos (int): Position.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.genome_index">
<span class="sig-name descname"><span class="pre">genome_index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ci</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strand</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.genome_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute genome index for a chromosome index and position.</p>
<dl class="simple">
<dt>Args</dt><dd><p>ci (int): Chromosome index.
pos (int): Position.
strand (str): ‘+’ or ‘-’</p>
</dd>
<dt>Returns:</dt><dd><p>gi (int): Genomic index</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.genome_indexes">
<span class="sig-name descname"><span class="pre">genome_indexes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ci</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positions</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strand</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.genome_indexes" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute genome indexes for a chromosome index and positions.</p>
<dl class="simple">
<dt>Args</dt><dd><p>ci (int): Chromosome index.
positions (int): Position(s)
strand (str): ‘+’ or ‘-’</p>
</dd>
<dt>Returns:</dt><dd><p>gi (int): Genomic index</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.genome_index_chrom">
<span class="sig-name descname"><span class="pre">genome_index_chrom</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strand</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.genome_index_chrom" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute genome index for a chromosome label and position.</p>
<dl class="simple">
<dt>Args</dt><dd><p>chrom (str): Chromosome label.
pos (int): Position.
strand (str): ‘+’ or ‘-’</p>
</dd>
<dt>Returns:</dt><dd><p>gi (int): Genomic index</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.genome_chr">
<span class="sig-name descname"><span class="pre">genome_chr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genome_indexes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.genome_chr" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Filter and convert an array of genome indexes</dt><dd><p>to indexes for a specific chromosome.</p>
</dd>
<dt>Args</dt><dd><p>genome_indexes (np.array):
chrom (str):</p>
</dd>
<dt>Returns</dt><dd><p>chrom_indexes (np.array)</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.learn_shift_pair">
<span class="sig-name descname"><span class="pre">learn_shift_pair</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.learn_shift_pair" title="Permalink to this definition">¶</a></dt>
<dd><p>Learn the optimal fragment shift from paired end fragments.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.learn_shift_single">
<span class="sig-name descname"><span class="pre">learn_shift_single</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">shift_min</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">shift_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">350</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.learn_shift_single" title="Permalink to this definition">¶</a></dt>
<dd><p>Learn the optimal fragment shift that maximizes across strand correlation</p>
<p>(to be applied for single end discordant alignments.)</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.read_bam">
<span class="sig-name descname"><span class="pre">read_bam</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_sorted</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_overlap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.read_bam" title="Permalink to this definition">¶</a></dt>
<dd><p>Read alignments from a BAM file into unique and multi data structures.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.infer_active_blocks">
<span class="sig-name descname"><span class="pre">infer_active_blocks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genome_coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_inactive</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.infer_active_blocks" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.infer_active_blocks_groupby">
<span class="sig-name descname"><span class="pre">infer_active_blocks_groupby</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genome_coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_inactive</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.infer_active_blocks_groupby" title="Permalink to this definition">¶</a></dt>
<dd><p>Infer active genomic blocks that we’ll need to consider.</p>
<dl class="simple">
<dt>The non-groupby version is inefficient. This one should improve it,</dt><dd><p>but it’s unfinished.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.read_multi_bwa">
<span class="sig-name descname"><span class="pre">read_multi_bwa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">multi_positions</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_reads</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_weight</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ri</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align_shift_forward</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align_shift_reverse</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.read_multi_bwa" title="Permalink to this definition">¶</a></dt>
<dd><p>Helper function to process a BWA multi-mapper.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.read_multi_nh">
<span class="sig-name descname"><span class="pre">read_multi_nh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">multi_positions</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_reads</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_weight</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ri</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">last_read_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_sorted</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multi_read_index</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.read_multi_nh" title="Permalink to this definition">¶</a></dt>
<dd><p>Helper function to process an NH-tagged multi-mapper.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.set_clips">
<span class="sig-name descname"><span class="pre">set_clips</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coverage</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.set_clips" title="Permalink to this definition">¶</a></dt>
<dd><p>Hash indexes to clip at various thresholds.</p>
<p>Must run this before running clip_multi, which will use
self.multi_clip_indexes. The objective is to estimate
coverage conservatively w/ clip_max and smoothing before
asking whether the raw coverage count is compelling.</p>
<dl class="simple">
<dt>In:</dt><dd><p>coverage (np.array): Pre-clipped genome coverage.</p>
</dd>
<dt>Out:</dt><dd><dl class="simple">
<dt>self.adaptive_t (int-&gt;float): Clip values mapped to coverage thresholds</dt><dd><p>above which to apply them.</p>
</dd>
<dt>self.multi_clip_indexes (int-&gt;np.array): Clip values mapped to genomic</dt><dd><p>indexes to clip.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.clip_multi">
<span class="sig-name descname"><span class="pre">clip_multi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.clip_multi" title="Permalink to this definition">¶</a></dt>
<dd><p>Clip coverage at adaptively-determined thresholds.</p>
<p>Must run set_clips to set self.multi_clip_indexes. If self.adaptive_t
is empty, I know it hasn’t been run yet.</p>
<dl class="simple">
<dt>In:</dt><dd><p>coverage (np.array): Pre-clipped genome/chromosome coverage.
chrom (str): Single chromosome coverage.</p>
</dd>
<dt>Out:</dt><dd><p>coverage (np.array): Clipped coverage values</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bam_cov.GenomeCoverage.write">
<span class="sig-name descname"><span class="pre">write</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">single_or_pair</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smooth_sd</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zero_eps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.003</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.GenomeCoverage.write" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute and write out coverage to bigwig file.</p>
<p>Go chromosome by chromosome here to facilitate printing,
and save memory.</p>
<dl class="simple">
<dt>In:</dt><dd><p>output_file (str): HDF5 or BigWig filename.
single_or_pair (bool): Specifies whether to correct for paired end double coverage.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.cigar_len">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">cigar_len</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cigar_str</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.cigar_len" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bam_cov.scatter_lims">
<span class="sig-prename descclassname"><span class="pre">bam_cov.</span></span><span class="sig-name descname"><span class="pre">scatter_lims</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vals1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vals2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bam_cov.scatter_lims" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../index.html">Init_website</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../index.html">API Reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, David Kelley, Gunalan Natesan.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 6.1.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../_sources/autoapi/bam_cov/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>