
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>basenji_sat_h5 &#8212; Init_website 1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css" />
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="module-basenji_sat_h5">
<span id="basenji-sat-h5"></span><h1><a class="reference internal" href="#module-basenji_sat_h5" title="basenji_sat_h5"><code class="xref py py-mod docutils literal notranslate"><span class="pre">basenji_sat_h5</span></code></a><a class="headerlink" href="#module-basenji_sat_h5" title="Permalink to this heading">¶</a></h1>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this heading">¶</a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this heading">¶</a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.main" title="basenji_sat_h5.main"><code class="xref py py-obj docutils literal notranslate"><span class="pre">main</span></code></a>()</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.enrich_activity" title="basenji_sat_h5.enrich_activity"><code class="xref py py-obj docutils literal notranslate"><span class="pre">enrich_activity</span></code></a>(seqs, seqs_1hot, targets, ...)</p></td>
<td><p>Filter data for the most active sequences in the set.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.expand_4l" title="basenji_sat_h5.expand_4l"><code class="xref py py-obj docutils literal notranslate"><span class="pre">expand_4l</span></code></a>(sat_lg_ti, seq_1hot)</p></td>
<td><p>Expand</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.delta_matrix" title="basenji_sat_h5.delta_matrix"><code class="xref py py-obj docutils literal notranslate"><span class="pre">delta_matrix</span></code></a>(seqs_1hot, sat_preds, satmut_len)</p></td>
<td><p>Compute the matrix of prediction deltas</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.loss_gain" title="basenji_sat_h5.loss_gain"><code class="xref py py-obj docutils literal notranslate"><span class="pre">loss_gain</span></code></a>(sat_delta, sat_preds_si, satmut_len)</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.parse_input" title="basenji_sat_h5.parse_input"><code class="xref py py-obj docutils literal notranslate"><span class="pre">parse_input</span></code></a>(input_file, sample)</p></td>
<td><p>Parse an input file that might be FASTA or HDF5.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.plot_heat" title="basenji_sat_h5.plot_heat"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_heat</span></code></a>(ax, sat_delta_ti, min_limit)</p></td>
<td><p>Plot satmut deltas.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.plot_predictions" title="basenji_sat_h5.plot_predictions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_predictions</span></code></a>(ax, preds, satmut_len, seq_len, buffer)</p></td>
<td><p>Plot the raw predictions for a sequence and target</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.plot_sad" title="basenji_sat_h5.plot_sad"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_sad</span></code></a>(ax, sat_loss_ti, sat_gain_ti)</p></td>
<td><p>Plot loss and gain SAD scores.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.plot_seqlogo" title="basenji_sat_h5.plot_seqlogo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_seqlogo</span></code></a>(ax, seq_1hot, sat_score_ti[, pseudo_pct])</p></td>
<td><p>Plot a sequence logo for the loss/gain scores.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#basenji_sat_h5.satmut_seqs" title="basenji_sat_h5.satmut_seqs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">satmut_seqs</span></code></a>(seqs_1hot, satmut_len)</p></td>
<td><p>Construct a new array with the given sequences and saturated</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#basenji_sat_h5.subplot_params" title="basenji_sat_h5.subplot_params"><code class="xref py py-obj docutils literal notranslate"><span class="pre">subplot_params</span></code></a>(seq_len)</p></td>
<td><p>Specify subplot layout parameters for various sequence lengths.</p></td>
</tr>
</tbody>
</table>
<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.main">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">main</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.main" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.enrich_activity">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">enrich_activity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqs_1hot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">targets</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">activity_enrich</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_indexes</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.enrich_activity" title="Permalink to this definition">¶</a></dt>
<dd><p>Filter data for the most active sequences in the set.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.expand_4l">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">expand_4l</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sat_lg_ti</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_1hot</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.expand_4l" title="Permalink to this definition">¶</a></dt>
<dd><p>Expand</p>
<dl class="simple">
<dt>In:</dt><dd><p>sat_lg_ti (l array): Sat mut loss/gain scores for a single sequence and
target.
seq_1hot (Lx4 array): One-hot coding for a single sequence.</p>
</dd>
<dt>Out:</dt><dd><p>sat_loss_4l (lx4 array): Score-hot coding?</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.delta_matrix">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">delta_matrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seqs_1hot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_preds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">satmut_len</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.delta_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the matrix of prediction deltas</p>
<dl>
<dt>Args:</dt><dd><p>seqs_1hot (Lx4 array): One-hot coding of all sequences.
sat_preds: (SMxLxT array): Satmut sequence predictions.
satmut_len: Saturated mutagenesis region length.</p>
</dd>
<dt>Returns:</dt><dd><p>sat_delta (4 x L_sm x T array): Delta matrix for saturated mutagenesis
region.</p>
</dd>
<dt>Todo:</dt><dd><p>-Rather than computing the delta as the change at that nucleotide’s
prediction,</p>
<blockquote>
<div><p>compute it as the mean change across the sequence. That way, we
better
pick up on motif-flanking interactions.</p>
</div></blockquote>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.loss_gain">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">loss_gain</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sat_delta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_preds_si</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">satmut_len</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.loss_gain" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.parse_input">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">parse_input</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.parse_input" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse an input file that might be FASTA or HDF5.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.plot_heat">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">plot_heat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_delta_ti</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_limit</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.plot_heat" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot satmut deltas.</p>
<dl class="simple">
<dt>Args:</dt><dd><p>ax (Axis): matplotlib axis to plot to.
sat_delta_ti (4 x L_sm array): Single target delta matrix for saturated mutagenesis region,
min_limit (float): Minimum heatmap limit.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.plot_predictions">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">plot_predictions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">preds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">satmut_len</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_len</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.plot_predictions" title="Permalink to this definition">¶</a></dt>
<dd><dl>
<dt>Plot the raw predictions for a sequence and target</dt><dd><p>across the specificed saturated mutagenesis region.</p>
</dd>
<dt>Args:</dt><dd><p>ax (Axis): matplotlib axis to plot to.
preds (L array): Target predictions for one sequence.
satmut_len (int): Satmut length from which to determine</p>
<blockquote>
<div><p>the values to plot.</p>
</div></blockquote>
<p>seq_len (int): Full sequence length.
buffer (int): Ignored buffer sequence on each side</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.plot_sad">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">plot_sad</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_loss_ti</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_gain_ti</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.plot_sad" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot loss and gain SAD scores.</p>
<dl class="simple">
<dt>Args:</dt><dd><p>ax (Axis): matplotlib axis to plot to.
sat_loss_ti (L_sm array): Minimum mutation delta across satmut length.
sat_gain_ti (L_sm array): Maximum mutation delta across satmut length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.plot_seqlogo">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">plot_seqlogo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_1hot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sat_score_ti</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudo_pct</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.plot_seqlogo" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot a sequence logo for the loss/gain scores.</p>
<dl class="simple">
<dt>Args:</dt><dd><p>ax (Axis): matplotlib axis to plot to.
seq_1hot (Lx4 array): One-hot coding of a sequence.
sat_score_ti (L_sm array): Minimum mutation delta across satmut length.
pseudo_pct (float): % of the max to add as a pseudocount.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.satmut_seqs">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">satmut_seqs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seqs_1hot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">satmut_len</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.satmut_seqs" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct a new array with the given sequences and saturated
mutagenesis versions of them.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="basenji_sat_h5.subplot_params">
<span class="sig-prename descclassname"><span class="pre">basenji_sat_h5.</span></span><span class="sig-name descname"><span class="pre">subplot_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq_len</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#basenji_sat_h5.subplot_params" title="Permalink to this definition">¶</a></dt>
<dd><p>Specify subplot layout parameters for various sequence lengths.</p>
</dd></dl>

</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../index.html">Init_website</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../index.html">API Reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, David Kelley, Gunalan Natesan.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 6.1.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../_sources/autoapi/basenji_sat_h5/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>